src/cds/src/main/kotlin/org/icpclive/cds/cats/CATSDataSource.kt

```diff
@@ -0,0 +1,175 @@
+//        )
+//        return ContestParseResult(contestInfo, listOf(), listOf())
+        return parseAndUpdateStandings(problemsLoader.loadOnce(), usersLoader.loadOnce(), contestLoader.loadOnce(), runsLoader.loadOnce())
+    }
+
+    private fun parseAndUpdateStandings(problems: Problems, users: Users, contest: Contest, runs: ArrayList<Run>): ContestParseResult {
+        val problemsList = ArrayList<ProblemInfo>()
+        val mapProblems = HashMap<Int, String>()
+        for (i in 0 until problems.problems.size) {
+            val problem = problems.problems[i]
+            mapProblems[problem.id] = problem.name
+            problemsList.add(
+                ProblemInfo(
+                    letter = problem.code,
+                    name = problem.name,
+                    color = Color(0x000000),
+                    id = problem.id,
+                    ordinal = i
+                )
+            )
+        }
+
+        val teamList = ArrayList<TeamInfo>()
+        val mapTeam = HashMap<Int, String>()
+        for (i in 0 until users.users.size) {
+            val team = users.users[i];
+            if (team.role != "in_contest") {
+                continue;
+            }
+            mapTeam[team.account_id] = team.name
+            teamList.add(
+                TeamInfo(
+                    id = team.account_id,
+                    name = team.name,
+                    shortName = team.name,
+                    contestSystemId = StringBuilder().append(team.account_id).toString(),
+                    groups = listOf(),
+                    hashTag = null,
+                    medias = mapOf()
+                )
+            )
+        }
+        // TODO: Add time.
+//        val startDate = LocalDateTime.parse(contest.start_date, DateTimeFormatter.ofPattern("dd.MM.yyyy HH:mm"))
+
+        val contestInfo = ContestInfo(
+            status = ContestStatus.OVER,
+            resultType = if (contest.rules == "icpc") ContestResultType.ICPC else ContestResultType.IOI,
+            // TODO: Add correct time
+            startTime = Clock.System.now(),
+            contestLength = 5.hours,
+            freezeTime = 1.hours,
+            problems = problemsList,
+            teams = teamList
+        )
+
+        val preparedRuns = runs.filter { it.type == "submit" && it.id != null }
+
+        val resultRuns = preparedRuns.map {
+            RunInfo(
+                id = it.id ?: 0,
+                ICPCRunResult(
+                    isAccepted = ("OK" == it.state_text),
+                    isAddingPenalty = ("OK" != it.state_text && "CE" != it.state_text),
+                    isFirstToSolveRun = false,
+                    result = it.state_text ?: ""
+                ),
+                problemId = it.problem_id ?: 0,
+                teamId = it.team_id ?: 0,
+                percentage = if ("" == it.state_text) 0.0 else 1.0,
+                time = 0.seconds
+            )
+        }
+
+        for (elem in resultRuns) {
```

Comment:
Please don't leave debug code. Or, if you think it's useful, you can use 
`val logger = getLogger(CATSDataSource::class)` and `logger.debug(...)`